modules/lang-painless/src/main/java/org/elasticsearch/painless/PainlessExecuteAction.java

```diff
@@ -86,31 +111,76 @@ public Response newResponse() {
 
         private static final ParseField SCRIPT_FIELD = new ParseField("script");
         private static final ParseField CONTEXT_FIELD = new ParseField("context");
-        private static final ConstructingObjectParser<Request, Void> PARSER = new ConstructingObjectParser<>(
+        private static final ParseField INDEX_FIELD = new ParseField("index");
+        private static final ParseField DOCUMENT_FIELD = new ParseField("document");
+        private static final ConstructingObjectParser<Request, ParseContext> PARSER = new ConstructingObjectParser<>(
             "painless_execute_request", args -> new Request((Script) args[0], (SupportedContext) args[1]));
 
+        private static class ParseContext {
+
+            private String index;
+            private BytesReference document;
+
+        }
+
         static {
             PARSER.declareObject(ConstructingObjectParser.constructorArg(), (p, c) -> Script.parse(p), SCRIPT_FIELD);
             PARSER.declareObject(ConstructingObjectParser.optionalConstructorArg(), (p, c) -> {
                 // For now only accept an empty json object:
                 XContentParser.Token token = p.nextToken();
                 assert token == XContentParser.Token.FIELD_NAME;
                 String contextType = p.currentName();
-                token = p.nextToken();
-                assert token == XContentParser.Token.START_OBJECT;
-                token = p.nextToken();
-                assert token == XContentParser.Token.END_OBJECT;
-                token = p.nextToken();
-                assert token == XContentParser.Token.END_OBJECT;
-                return SupportedContext.valueOf(contextType.toUpperCase(Locale.ROOT));
+                SupportedContext supportedContext = SupportedContext.valueOf(contextType.toUpperCase(Locale.ROOT));
+                if (supportedContext == SupportedContext.PAINLESS_TEST) {
+                    token = p.nextToken();
+                    assert token == XContentParser.Token.START_OBJECT;
+                    token = p.nextToken();
+                    assert token == XContentParser.Token.END_OBJECT;
+                    token = p.nextToken();
+                    assert token == XContentParser.Token.END_OBJECT;
+                } else if (supportedContext == SupportedContext.FILTER_SCRIPT || supportedContext == SupportedContext.SEARCH_SCRIPT) {
+                    for (token = p.nextToken(); token != XContentParser.Token.END_OBJECT; token = p.nextToken()) {
```

Comment:
Can you add an example, preferably in the docs for this, of what the input will look like here? It is difficult to figure out from the code. I also wonder if it would be easier to use object parser internally here?

Reply:
I still need to add docs, but it going to look like the examples I provided in the PR description:

```
"search_script": {
      "document": {
        "field": "four"
      },
      "index": "my-index"
    }
```

In this context case, a document and an index name needs to be provided.